projects/storefrontlib/cms-components/product/carousel/product-carousel/product-carousel.component.spec.ts

```diff
@@ -157,9 +158,23 @@ describe('ProductCarouselComponent', () => {
   it(
     'should have 2 items',
     waitForAsync(() => {
-      let items: Observable<Product>[];
-      component.items$.subscribe((i) => (items = i));
-      expect(items.length).toBe(2);
+      const productService = TestBed.inject(ProductService);
+      spyOn(productService, 'get').and.callThrough();
+
+      const scopes = [
+        ProductScope.LIST,
+        ProductScope.PRICE,
+        ProductScope.DETAILS,
+      ];
+
+      let items: Observable<Product>[] | undefined;
+      component.items$.subscribe((i) => {
+        expect(productService.get).toHaveBeenCalledTimes(2);
+        expect(productService.get).toHaveBeenCalledWith('1', scopes);
+        expect(productService.get).toHaveBeenCalledWith('2', scopes);
+        items = i;
```

Comment:
Do you need the done() function for assertions in a subscription?

Reply:
Ah, yes, I think there should be a done(). Thanks for reminding me.

Reply:
Actually, line 177 essentially checks that lines 171-175 are called, but I like the neater approach of putting all assertions in the callback.